### ISSUE TITLE
Can't run quickstart on DigitalOcean

### ISSUE THREAD

**Issue by jonatron** [neuralmagic/deepsparse#128]
**Describe the bug**
Can't run quickstart https://github.com/neuralmagic/deepsparse#quickstart-with-sparsezoo-onnx-models on DigitalOcean droplet with "Premium Intel" CPU

**Expected behavior**
No error message

**Environment**
1. OS Ubuntu 20.04 (LTS) x64
2. Python version 3.8
3. DeepSparse version 0.5.1

**To Reproduce**
Exact steps to reproduce the behavior:
```
apt update
apt-get -y install python3-venv build-essential cmake
python3 -m venv venv
source venv/bin/activate
pip install --upgrade pip setuptools wheel
pip install deepsparse
```
Run "ResNet-50 Dense" example linked above

**Errors**
`OSError: neuralmagic: encountered exception while trying read arch.bin: Command '/root/venv/lib/python3.8/site-packages/deepsparse/arch.bin' returned non-zero exit status 1.`

**Additional context**
The binary doesn't output anything:
```

(venv) root@ubuntu-s-1vcpu-2gb-intel-lon1-01:~# /root/venv/lib/python3.8/site-packages/deepsparse/arch.bin
(venv) root@ubuntu-s-1vcpu-2gb-intel-lon1-01:~# 
```

**Comment by mgoin** [comment]
Hi @jonatron thanks for the bug report! We had trouble replicating this until another user reported similar issues with shared CPU DigitalOcean droplets.

`arch.bin` is used by DeepSparse to get information about the system's hardware topology so the engine can decide what algorithms to use and make optimizations. One of the key features needed is information about the number of cores and how much L3 cache is available. Looking at the `cpuid` on one of these DigitalOcean droplets shows us that the L3 cache is missing/disabled:

```
   L3 cache information (0x80000006/edx):
      line size (bytes)     = 0x0 (0)
      lines per tag         = 0x0 (0)
      associativity         = L2 off (0)
      size (in 512KB units) = 0x0 (0)
```

There is also limited information about what threads are on what physical cores and their properties, which is needed to schedule work efficiently.

So unfortunately while we have an idea of why the system is in the state it is (something like [Firecracker](https://firecracker-microvm.github.io/) comes to mind), we don't have a fix for this quite yet. We hope to support accelerated inference on this low-cost hardware in the near future.

**Comment by jonatron** [comment]
Thanks for the update!

I can deduce the physical core architecture:
```
root@ubuntu-s-1vcpu-2gb-intel-lon1-01:~# cat /sys/devices/cpu/caps/pmu_name
skylake
```

According to https://en.wikichip.org/wiki/intel/microarchitectures/skylake_(server) 
L1I Cache | 32 KiB/core8-way set associative
L1D Cache | 32 KiB/core8-way set associative
L2 Cache | 1 MiB/core16-way set associative
L3 Cache | 1.375 MiB/core11-way set associative

Of course that can't be detected automatically. Perhaps the CPU details could be manually set via an environment variable or some other way?

Also, if it's detecting 4MiB L2 cache:
```
root@ubuntu-s-1vcpu-2gb-intel-lon1-01:~# cat /proc/cpuinfo 
cache size	: 4096 KB
root@ubuntu-s-1vcpu-2gb-intel-lon1-01:~# cat /sys/devices/system/cpu/cpu0/cache/index2/size 
4096K
```
where I've only got 1 core in my VM, should it actually use 1MiB?

**Comment by mgoin** [comment]
@jonatron yes I believe the best way past this in the short-term is to set the CPU details manually. We're working on getting an accessible way to do this but unfortunately isn't trivial right now. I will keep you in the loop when we get this released.

Correct, it "should" be reporting 1MiB L2 cache since you are only on 1 core. Specifically you might only be running on one hyperthread since it is `1vcpu`, so something weird could be setup by DO here.
